### ISSUE TITLE
Dropout and LSTM

### ISSUE THREAD

**Issue by twoletters** [zacwellmer/WorldModels#7]
Hi Zac,

It looks like the dropout features in the LSTM layer are not used at all. According to the comments in the code, that might not be intended:
`rnn_out, h, c = rnn.inference_base(input_x, initial_state=states, training=training) # set training True to use Dropout`
For this to work, the `dropout` or/and `recurrent_dropout` parameters need to be specified when the LSTM layer is created in `rnn/rnn.py` on line 28:
`self.inference_base = tf.keras.layers.LSTM(units=args.rnn_size, return_sequences=True, return_state=True, time_major=False, dropout=args.rnn_dropout, recurrent_dropout=args.rnn_rec_dropout)`
with both `args.rnn_dropout` and `args.rnn_rec_dropout` around 0.4 as a starting point.

If the dropouts are not required, then the LSTM layer could be replaced with `keras.layers.CuDNNLSTM`, which is much faster to train on a GPU. The full requirements to replace the pure Tensorflow LSTM implementation with the CuDNN implementation are listed towards the top of [this page](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LSTM).

Dropouts or speed, which will you choose?

**Comment by zacwellmer** [comment]
That comment was left over from some preliminary experiments. I think Keras uses the CuDNNLSTM under the hood when dropout is off

> If a GPU is available and all the arguments to the layer meet the requirement of the CuDNN kernel (see below for details), the layer will use a fast cuDNN implementation
-[from the docs](https://www.tensorflow.org/versions/r2.2/api_docs/python/tf/keras/layers/LSTM)

**Comment by twoletters** [comment]
Yes, I think you are right. You won't use the dropouts in the training, I conclude.
